src/lib.rs

```diff
@@ -216,19 +216,19 @@ pub trait ThirtyTwoByteHash {
 #[cfg(feature = "bitcoin_hashes")]
 #[cfg_attr(docsrs, doc(cfg(feature = "bitcoin-hashes")))]
 impl ThirtyTwoByteHash for hashes::sha256::Hash {
-    fn into_32(self) -> [u8; 32] { self.into_inner() }
+    fn into_32(self) -> [u8; 32] { self.into_byte_array() }
```

Comment:
Why it's `into_byte_array` and not `to_byte_array`?

Reply:
Because I could not work out the rules for `Copy` types from the conventions table so I assumed we had it correct from last time we tried :)

Reply:
Although since this PR is specifically fixing misnaming of `as_hash` perhaps I should not have trusted old us.

Reply:
Oh wait, its owned -> owned `Copy` type  - that means `to_byte_array`. I must have been confused with some other time I was reading that table recently.

Reply:
Same argument for `to_raw_hash` - face palm.

Reply:
Basically:

* If it's reference-to-reference => `as_`
* If the input value was invalidated by move => `into_`
* If the input value is usable after the call => `to_`
* In generic code where it's not clear whether the input value is `Copy` or not => `into_`

Reply:
Oh sweeeet, that's way better than that damn table in the conventions doc. Thanks bro!